lib/Dialect/CKKS/IR/CKKSOps.td

```diff
@@ -0,0 +1,198 @@
+#ifndef THIRD_PARTY_HEIR_LIB_DIALECT_CKKS_IR_CKKSOPS_TD_
+#define THIRD_PARTY_HEIR_LIB_DIALECT_CKKS_IR_CKKSOPS_TD_
+
+include "CKKSDialect.td"
+
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+
+include "lib/Dialect/LWE/IR/LWETypes.td"
+include "lib/Dialect/LWE/IR/LWETraits.td"
+include "mlir/Dialect/Polynomial/IR/PolynomialAttributes.td"
+
+class CKKS_Op<string mnemonic, list<Trait> traits = []> :
+        Op<CKKS_Dialect, mnemonic, traits> {
+  let cppNamespace = "::mlir::heir::ckks";
+  let assemblyFormat = [{
+    operands attr-dict `:`  functional-type(operands, results)
+  }];
+}
+
+class CKKS_CiphertextPlaintextOp<string mnemonic, list<Trait> traits =
+    [AllTypesMatch<["ciphertextInput", "output"]>,
+    TypesMatchWith<"type of 'plaintextInput' matches encoding type of 'ciphertextInput'",
+                    "ciphertextInput", "plaintextInput",
+                    "lwe::RLWEPlaintextType::get($_ctxt, "
+                      "::llvm::cast<lwe::RLWECiphertextType>($_self).getEncoding(),"
+                      "::llvm::cast<lwe::RLWECiphertextType>($_self).getRlweParams().getRing(),"
+                      "::llvm::cast<lwe::RLWECiphertextType>($_self).getUnderlyingType())">]> :
+        CKKS_Op<mnemonic, traits> {
+  let arguments = (ins
+    RLWECiphertext:$ciphertextInput,
+    RLWEPlaintext:$plaintextInput
+  );
+
+  let results = (outs
+    RLWECiphertext:$output
+  );
+}
+
+def CKKS_AddOp : CKKS_Op<"add", [Commutative, SameOperandsAndResultType]> {
```

Comment:
not your fault since i don't think bgv has these either but i believe these are all Pure ops as well